app/src/main/java/org/jellyfin/androidtv/preferences/UserPreferences.kt

```diff
@@ -0,0 +1,169 @@
+package org.jellyfin.androidtv.preferences
+
+import android.content.Context
+import android.preference.PreferenceManager
+import org.jellyfin.androidtv.preferences.enums.AudioBehavior
+import org.jellyfin.androidtv.preferences.enums.LoginBehavior
+import org.jellyfin.androidtv.preferences.enums.PreferredVideoPlayer
+
+/**
+ * User preferences are configurable by the user and change behavior of the application.
+ * When changing preferences migration should be added to the init function.
+ *
+ * @param context Context to get the SharedPreferences from
+ */
+class UserPreferences(context: Context) : SharedPreferenceStore(PreferenceManager.getDefaultSharedPreferences(context)) {
+	/* Authentication */
+	/**
+	 * Behavior for login when starting the app.
+	 * **note**: Currently settable via user-preferences only due too custom logic
+	 */
+	var loginBehavior by enumPreference("login_behavior", LoginBehavior.SHOW_LOGIN)
+		private set
+
+	/**
+	 * Ask for password when starting the app
+	 */
+	var passwordPromptEnabled by booleanPreference("pref_auto_pw_prompt", false)
+
+	/**
+	 * Use login using pin (when set)
+	 */
+	var passwordDPadEnabled by booleanPreference("pref_alt_pw_entry", false)
+
+	/**
+	 * Sign out automatically after x milliseconds
+	 */
+	var autoSignOutTimeout by stringPreference("pref_auto_logoff_timeout", "3600000")
+
+	/* Display */
+	/**
+	 * Enable background images while browsing
+	 */
+	var backdropEnabled by booleanPreference("pref_show_backdrop", false)
```

Comment:
It seems all boolean preferences are being defaulted to `false` now. Is this correct?

Reply:
I might have forgotten to set the default values for the booleans. I'll check them tomorrow.